eth/handler.go

```diff
@@ -295,6 +295,13 @@ func (pm *ProtocolManager) handle(p *peer) error {
 			glog.V(logger.Warn).Infof("%v: timed out DAO fork-check, dropping", p)
 			pm.removePeer(p.id)
 		})
+		// Make sure it's cleaned up if the peer dies off
+		defer func() {
+			if p.forkDrop != nil {
```

Comment:
The reply to the challenge in:

 * On assumed success: https://github.com/ethereum/go-ethereum/blob/develop/eth/handler.go#L515
 * On verified success: https://github.com/ethereum/go-ethereum/blob/develop/eth/handler.go#L526